Validation/MuonHits/src/RPCSimHitMatcher.cc

```diff
@@ -0,0 +1,160 @@
+#include "Validation/MuonHits/interface/RPCSimHitMatcher.h"
+#include "Geometry/CommonTopologies/interface/TrapezoidalStripTopology.h"
+
+using namespace std;
+
+RPCSimHitMatcher::RPCSimHitMatcher(const edm::ParameterSet& ps,
+                                   edm::ConsumesCollector&& iC)
+    : MuonSimHitMatcher(ps, std::move(iC)) {
+  simHitPSet_ = ps.getParameterSet("rpcSimHit");
+  verbose_ = simHitPSet_.getParameter<int>("verbose");
+  simMuOnly_ = simHitPSet_.getParameter<bool>("simMuOnly");
+  discardEleHits_ = simHitPSet_.getParameter<bool>("discardEleHits");
+
+  simHitInput_ = iC.consumes<edm::PSimHitContainer>(
+      simHitPSet_.getParameter<edm::InputTag>("inputTag"));
+}
+
+/// initialize the event
+void RPCSimHitMatcher::init(const edm::Event& iEvent,
+                            const edm::EventSetup& iSetup) {
+  iSetup.get<MuonGeometryRecord>().get(rpc_geom_);
+  if (rpc_geom_.isValid()) {
+    geometry_ = &*rpc_geom_;
+  } else {
+    hasGeometry_ = false;
+    edm::LogWarning("RPCSimHitMatcher")
+        << "+++ Info: RPC geometry is unavailable. +++\n";
+  }
+  MuonSimHitMatcher::init(iEvent, iSetup);
+}
+
+/// do the matching
+void RPCSimHitMatcher::match(const SimTrack& track, const SimVertex& vertex) {
+  MuonSimHitMatcher::match(track, vertex);
+
+  if (hasGeometry_) {
+    matchSimHitsToSimTrack(track_ids_, simHits_);
+
+    if (verbose_) {
+      edm::LogInfo("RPCSimHitMatcher")
+          << "nSimHits " << simHits_.size() << " nTrackIds "
+          << track_ids_.size() << endl;
+      edm::LogInfo("RPCSimHitMatcher")
+          << "detids RPC " << detIds().size() << endl;
+
+      const auto& ch_ids = chamberIds();
+      for (const auto& id : ch_ids) {
+        const auto& simhits = MuonSimHitMatcher::hitsInChamber(id);
+        const auto& simhits_gp = simHitsMeanPosition(simhits);
+        edm::LogInfo("RPCSimHitMatcher")
+            << "RPCDetId " << RPCDetId(id) << ": nHits " << simhits.size()
+            << " eta " << simhits_gp.eta() << " phi " << simhits_gp.phi()
+            << " nCh " << chamber_to_hits_[id].size() << endl;
+        const auto& strips = hitStripsInDetId(id);
+        edm::LogInfo("RPCSimHitMatcher") << "nStrips " << strips.size() << endl;
+        edm::LogInfo("RPCSimHitMatcher") << "strips : ";
+        std::copy(strips.begin(), strips.end(),
+                  ostream_iterator<int>(std::cout, " "));
```

Comment:
This will also write to `cout` which is forbidden. Could you please substitute this call with `MessageLogger` as well?

Reply:
Done.